app/src/main/java/com/expidev/gcmapp/MeasurementDetailsActivity.java

```diff
@@ -613,4 +808,48 @@ protected void onPostExecute(MeasurementDetails measurementDetails)
             onLoadDetailsFromServer(measurementDetails);
         }
     }
+
+    private class SaveMeasurementsToServer extends AsyncTask<MeasurementDetails, Void, Boolean>
+    {
+        @Override
+        protected Boolean doInBackground(MeasurementDetails... params)
+        {
+            MeasurementDetails measurementDetails = params[0];
+            GmaApiClient apiClient = GmaApiClient.getInstance(MeasurementDetailsActivity.this);
+
+            try
+            {
+                List<MeasurementDetails> data = new ArrayList<>();
+                data.add(measurementDetails);
+
+                Log.i(TAG, "Saving new measurements to the database");
+                MeasurementsService.saveMeasurementDetailsToDatabase(MeasurementDetailsActivity.this, measurementDetails);
+
+                Log.i(TAG, "Sending new measurements to the server");
+                return apiClient.updateMeasurementDetails(data, preferences.getString(PREF_CURRENT_ASSIGNMENT, null));
```

Comment:
also of note (and @jonvellacott can speak into this), ministries that you are an inherited_leader for don't have an assignment_id in the response JSON. I'm not sure if there are execution paths that hit this corner case or not. So, the assignment_id is potentially null.